src/amazonqGumby/telemetry/codeTransformTelemetry.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This class contains helper methods dedicated to logging metrics specific to
+ * CodeTransform
+ */
+
+import { CodeTransformStartSrcComponents, telemetry } from '../../shared/telemetry/telemetry'
+import { JDKVersion } from '../../codewhisperer/models/model'
+import * as CodeWhispererConstants from '../../codewhisperer/models/constants'
+import { codeTransformTelemetryState } from './codeTransformTelemetryState'
+import { CodeTransformConstants } from '../constants'
+
+export class CodeTransformTelemetry {
+    public static SidePanelTransformTreeNode = 'devToolsStartButton'
+    public static HubStartButton = 'bottomPanelSideNavButton'
+
+    static logCodeTransformInitiatedMetric(source: string): void {
+        const commonMetrics = {
+            codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
+        }
+
+        if (source === CodeWhispererConstants.transformTreeNode) {
+            telemetry.codeTransform_isDoubleClickedToTriggerUserModal.emit({
+                codeTransformStartSrcComponents:
+                    CodeTransformTelemetry.SidePanelTransformTreeNode as CodeTransformStartSrcComponents,
+                ...commonMetrics,
+            })
+        } else if (source === CodeTransformConstants.HubStartButton) {
+            telemetry.codeTransform_isDoubleClickedToTriggerUserModal.emit({
+                codeTransformStartSrcComponents:
+                    CodeTransformTelemetry.HubStartButton as CodeTransformStartSrcComponents,
+                ...commonMetrics,
+            })
+        }
+
+        // todo: log metric for started from transform command
```

Comment:
Is it worth getting this in before merge?

Reply:
This is an out-of-date comment -- said metric is now being logged. I'll make sure this comment is removed